boards/shields/wiznet_wiz550io/wiznet_wiz550io.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024 Grant Ramsay
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This driver provides the initialsation sequence required for
+ * WIZ550io to configure the W5500 Ethernet controller with its
+ * embedded unique MAC address.
+ */
+
+#define DT_DRV_COMPAT wiznet_wiz550io
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(eth_w5500, CONFIG_ETHERNET_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/net/ethernet.h>
+
+#include "eth_w5500_priv.h"
+
+struct wiz550io_config {
+	const struct device *w5500_dev;
+	struct gpio_dt_spec reset;
+	struct gpio_dt_spec ready;
+};
+
+/* WIZ550io documentation recommends 150ms delay after HW reset
+ * for PIC12F519 MCU to configure the W5500.
+ */
+#define WIZ550IO_RESET_DELAY K_MSEC(150)
+
+static int wiz550io_init(const struct wiz550io_config *config)
```

Comment:
this could be added as a quirk to the original driver

Reply:
This has been discussed in both the dev-review and arch working group and has been agreed upon to have this as an exception for a shield, rather than a driver.

Reply:
are meeting replacing reviews now? I strongly disagree in this case.

Reply:
> are meetings replacing reviews now? 

No, and I never claimed otherwise. I merely stated it, the whole reason #75310 is merged, is to allow these exceptions.

Reply:
I think this can be solved properly without recurring to shield-level code. The shield-level binding alone just doesn't make any sense.

Reply:
@gmarull is it the existence of the binding itself that doesnt make sense or the location of it?

Reply:
> I think this can be solved properly without recurring to shield-level code

I think it could be solved without shield level code, by handling it in the application, but, this shield code has a benefit of the application not having to handle initialization of the module by providing this reusable code here. I don't see how it is really any different from zephyr handling initialization of many other things in various drivers and platform files. You could write a whole application without using Zephyr, but then what are we doing here?